docs/vsphere.md

```diff
@@ -1,23 +1,267 @@
 # VMware vSphere
 
+## Supported versions
+
+* 6.5
+* 6.7
+
+## Template VMs preparation
+
 To use the machine-controller to create machines on VMWare vsphere, you must first
-create a template.
+create a VM to be used as a template.
+
+*Note that:* 
+`template VMs` in this document refers to regular VMs and not
+[VM Templates][vm_templates] according to vSphere terminology.
+The difference is quite subtle, but VM Templates are not supported yet by
+`machine controller`.
 
-Ubuntu & CoreOS:
+### Create template VM from OVA
 
-1. Go into the VSphere WebUI, select your datacenter, right click onto it and choose "Deploy OVF Template"
-2. Fill in the "URL" field with the appropriate url:
-    * Ubuntu: `https://cloud-images.ubuntu.com/releases/18.04/release/ubuntu-18.04-server-cloudimg-amd64.ova`
-    * Container Linux: `https://stable.release.core-os.net/amd64-usr/current/coreos_production_vmware_ova.ova`
+To see where to locate the OVAs go to the OS specific section.
+
+#### WebUI procedure
+
+1. Go into the vSphere WebUI, select your datacenter, right click onto it and choose "Deploy OVF Template"
+2. Fill in the "URL" field with the appropriate url pointing to the `OVA` file
 3. Click through the dialog until "Select storage"
 4. Select the same storage you want to use for your machines
 5. Select the same network you want to use for your machines
 6. Leave everyhting in the "Customize Template" and "Ready to complete" dialog as it is
 7. Wait until the VM got fully imported and the "Snapshots" => "Create Snapshot" button is not grayed out anymore
 
-CentOS:
+#### Command-line procedure
+
+Prerequisites:
+
+* [GOVC](https://github.com/vmware/govmomi/tree/master/govc): tested on version 0.22.1
+* [jq](https://stedolan.github.io/jq/)
+
+Procedure:
+
+1. Download the `OVA` for the targeted OS.
+
+    ```
```

Comment:
* backticks
* no syntax hint
* indent

plus bunch of same blocks

Reply:
> * backticks
this is consistently used in the whole document
> * no syntax hint
no synthax hint because here we have some commands, I could put `bash` but does not provide much value as it is not something structured like `json`, `yaml`, etc.
> * indent
because we are inside a numbered list
> 
> plus bunch of same blocks
The same apply for all other blocks